app/src/main/java/com/example/returnpals/composetools/PricingUI.kt

```diff
@@ -26,90 +29,148 @@ import androidx.compose.ui.text.withStyle
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.lifecycle.ViewModel
+import androidx.navigation.NavController
+import com.example.returnpals.PricingPlan
+import com.example.returnpals.ScheduleReturn
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 
-// TODO: Guest UI
+// TODO: guest message "sign up for more pricing options"
 
 /////////////////////////////////////////////////////////////////////////////
 // PUBLIC API
 ////////////////////
 
-enum class Plan {
-    NONE, BRONZE, SILVER, GOLD, PLATINUM
+/**
+ * Draws the entire screen of the step "Choose PricingPlan" in the "Schedule a Return" process
+ */
+@Composable
+fun ScheduleReturn.PricingUI(
+    modifier: Modifier = Modifier,
+    navController: NavController? = null,
+    onChangePlan: (PricingPlan) -> Unit = {},
+    plan: PricingPlan? = null,
+    guest: Boolean = false,
+) {
+    ScheduleReturnScaffold(
+        step = 3,
+        onClickNext = { /*TODO: navigate to package details */ },
```

Comment:
I still feel like the function creates content and the click action events could be passed along as parameters. I think if you have the NavController like you do, you could still navigate. I am curious to see how it works.

Reply:
If this function didn't handle navigation then the function above it (that calls it) would have to. I haven't created that top-level function yet so I may just expose the onclicks like you advise. I'm still wondering how to go about that, I think the top-level function would be the function made by Gus in `mainmenu/pricing.kt` called `PricingContent`. I'll do this for the next merge.